clang/lib/Analysis/LifetimeSafety/FactsGenerator.cpp

```diff
@@ -358,13 +513,13 @@ llvm::SmallVector<Fact *> FactsGenerator::issuePlaceholderLoans() {
 
   llvm::SmallVector<Fact *> PlaceholderLoanFacts;
   for (const ParmVarDecl *PVD : FD->parameters()) {
-    if (hasOrigin(PVD)) {
-      const PlaceholderLoan *L =
-          FactMgr.getLoanMgr().createLoan<PlaceholderLoan>(PVD);
-      OriginID OID = FactMgr.getOriginMgr().getOrCreate(*PVD);
-      PlaceholderLoanFacts.push_back(
-          FactMgr.createFact<IssueFact>(L->getID(), OID));
-    }
+    OriginList *List = getOriginsList(*PVD);
+    if (!List)
+      continue;
+    const PlaceholderLoan *L =
+        FactMgr.getLoanMgr().createLoan<PlaceholderLoan>(PVD);
+    PlaceholderLoanFacts.push_back(
```

Comment:
What is the main reason to not create placeholder loans for the nested origins?

Reply:
This directly effects annotation suggestions. We could do that but we would need to silence the annotation suggestions for inner origins because lifetimebound cannot express those.